pandas/_libs/groupby_helper.pxi.in

```diff
@@ -588,7 +588,13 @@ def group_rank_{{name}}(ndarray[float64_t, ndim=2] out,
                 if out[i, 0] != out[i, 0] or out[i, 0] == NAN:
                     out[i, 0] = NAN
                 else:
-                    out[i, 0] = out[i, 0] / grp_sizes[i, 0]
+                    if grp_sizes[i, 0] != 0:
+                        # BUG 23666 - tie-breaking for dense causes the
```

Comment:
I think this comment would be better served if shortened and placed before the condition rather than within